naivePhysics.js

```diff
@@ -233,11 +237,26 @@ export class NaivePhysics {
   #handleChildren(key, xBefore, xAfter, yBefore, yAfter) {
     const xDelta = xAfter - xBefore;
     const yDelta = yAfter - yBefore;
+    const options = this.objects[key].options;
     const children = this.objects[key].children;
     children.forEach((child) => {
-      child.options.x += xDelta;
-      child.options.y += yDelta;
-      this.#redrawObject(child);
+      if (xDelta) child.options.x += xDelta;
+      if (yDelta) child.options.y += yDelta;
+
+      const canRotate =
+        options.rotation &&
+        options.rotationX !== NaN &&
+        options.rotationY !== NaN;
+      if (canRotate) {
+        // rotate child with its (x,y) coordinates being relative to its parent's (x,y):
```

Comment:
note: children in a group don't get their rotation based off their relative position to their parent, but rather their position relative to the global axis (the top-left corner), which is then translated to be relative to their parent